assignment/core/src/main/java/com/assignment/core/models/BreadcrumbModel.java

```diff
@@ -0,0 +1,66 @@
+package com.assignment.core.models;
+
+import com.day.cq.wcm.api.Page;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.models.annotations.DefaultInjectionStrategy;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.ScriptVariable;
+import org.apache.sling.models.annotations.injectorspecific.Self;
+
+import javax.annotation.PostConstruct;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+@Model(adaptables = SlingHttpServletRequest.class, defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
+public class BreadcrumbModel {
+
+    @Self
+    private SlingHttpServletRequest request;
+
+    @ScriptVariable
+    private Page currentPage;
+
+    private List<BreadcrumbItem> breadcrumbs;
+
+    @PostConstruct
+    protected void init() {
+        breadcrumbs = new ArrayList<>();
+
+        Page parent = currentPage.getParent();
+        if (parent != null && !"/content".equals(parent.getPath())) {
+            breadcrumbs.add(new BreadcrumbItem(parent.getTitle(), parent.getPath()));
+
+            Page grandparent = parent.getParent();
+            if (grandparent != null && !"/content".equals(grandparent.getPath())) {
+                breadcrumbs.add(new BreadcrumbItem(grandparent.getTitle(), grandparent.getPath()));
+            }
+        }
+
+
+        Collections.reverse(breadcrumbs);
```

Comment:
Why did you reverse this list?

Reply:
because in the list i will get the parent and grandparent from the current node so if i dont reverse it i will get , en/us/assignment instead of assignment/us/en

> Why did you reverse this list?